pkg/component/controller/extensions_controller.go

```diff
@@ -515,7 +518,9 @@ func (ec *ExtensionsController) startControllerManager(ctx context.Context) {
 // Stop
 func (ec *ExtensionsController) Stop() error {
 	ec.L.Info("Stopping extensions controller")
-	ec.mgrCancelFn()
+	if ec.mgrCancelFn != nil {
+		ec.mgrCancelFn()
+	}
```

Comment:
It's probably good to add the mutex here, so there are no data races, as this is being used by multiple goroutines without any guarantees about concurrency.

```suggestion
	ec.mux.Lock()
	mgrCancelFn := ec.mgrCancelFn
	ec.mux.Unlock()
	if mgrCancelFn != nil {
		mgrCancelFn()
	}
```

Reply:
oh, yeah. Thanks for poiinting that out. Added mutex shielding for both places where `ec.mgrCancelFn` was called from